### PR TITLE
MEPTS 1240 | MQ Cat 9 Changes for June Release 2023 

### PR DESCRIPTION
**Pull request by Emmanuel0077**
…and pregnant queries

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 1]
```diff
@@ -8351,10 +8351,10 @@ public CohortDefinition getMQ9Num(int flag) {
 
     if (flag == 1 || flag == 3) {
       cd.setCompositionString(
-          "A AND requestCd4 AND NOT (C OR D OR E OR pregnantOnPeriod OR breastfeedingOnPeriod) AND AGE");
+          "A AND D AND breastfeedingOnPeriod AND requestCd4 AND NOT (C OR E OR pregnantOnPeriod) AND AGE");
     } else if (flag == 2 || flag == 4) {
       cd.setCompositionString(
-          "A AND resultCd4 AND NOT (C OR D OR E OR pregnantOnPeriod OR breastfeedingOnPeriod) AND AGE");
+          "A AND breastfeedingOnPeriod AND D AND resultCd4 AND NOT (C OR E OR pregnantOnPeriod) AND AGE");
```
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: This changes should not be applied on the numerator, please revert the changes and apply accordingly

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 2]
```diff
@@ -11490,29 +11499,41 @@ public CohortDefinition getMOHPregnantORBreastfeedingOnClinicalConsultation(
...
+                    + "       GROUP  BY p.patient_id) AS primeira "
+                    + "      inner join encounter enc "
+                    + "              ON enc.patient_id = primeira.patient_id "
+                    + "      inner join obs o "
+                    + "              ON o.encounter_id = enc.encounter_id "
+                    + "      inner join person pe "
+                    + "              ON pe.person_id = primeira.patient_id "
+                    + " WHERE  primeira.first_consultation >= :startDate "
+                    + "       AND primeira.first_consultation <= :endDate "
+                    + "       AND o.concept_id = ${question} "
```
Comment (helderjosue) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Please write the conditions in a more readable way

**Review by helderjosue — CHANGES_REQUESTED** [review]
Please check the comments

**Review by helderjosue — APPROVED** [review]
Looks good to me. Ready for final review

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 3]
```diff
@@ -7920,7 +7920,7 @@ public CohortDefinition getMQ9Den(int flag) {
             inclusionPeriodMappings));
 
     cd.setCompositionString(
-        "A AND NOT (C OR D OR E OR pregnantOnPeriod OR breastfeedingOnPeriod) AND AGE");
+        "A OR (D AND breastfeedingOnPeriod) AND NOT (C OR E OR pregnantOnPeriod) AND AGE");
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Does not meet the requirements

**Review by eliazmuchave — CHANGES_REQUESTED** [review]

**Review by eliazmuchave — APPROVED** [review]
Looks good to me

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 4]
```diff
@@ -11489,7 +11490,9 @@ public CohortDefinition getPatientsOnMQCat18Numerator() {
    *     Fim Revisão” menos 9 (-) meses. Nota 1: é a primeira consulta clínica de sempre do utente
    *     que decorreu no período de inclusão.
    * <li>Nota 2: A mulher grávida e lactante ao mesmo tempo, ou seja com registo de “Grávida=Sim” e
-   *     “Lactante=Sim” na mesma consulta inicial, será considerada como grávida.
+   *     “Lactante=Sim” na mesma consulta inicial, será considerada como grávida. <<<<<<< HEAD
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Resolve this conflicts

**Review by eliazmuchave — APPROVED** [review]
Looks good to me

**Pull request merged**